src/app/server/Mdns.cpp

```diff
@@ -119,31 +119,19 @@ CHIP_ERROR AdvertiseOperational()
     return mdnsAdvertiser.Advertise(advertiseParameters);
 }
 
-/// Set MDNS commissioner advertisement
-CHIP_ERROR AdvertiseCommisioner()
+/// Overloaded utility method for commissioner and commissionable advertisement
+// This method is used for both commissioner discovery and commissionable node discovery since
+// they share many fields.
+CHIP_ERROR Advertise(bool commissionableNode, CommissioningMode mode)
 {
-    return Advertise(false);
-}
+    bool commissioningMode = (mode == CommissioningMode::kEnabled || mode == CommissioningMode::kEnabledAsAdditionalCommissioning);
+    bool additionalCommissioning = (mode == CommissioningMode::kEnabledAsAdditionalCommissioning);
 
-/// Set MDNS commissionable node advertisement
-CHIP_ERROR AdvertiseCommissionableNode()
-{
-    return Advertise(true);
-}
-
-/// commissionableNode
-// CHIP_ERROR Advertise(chip::Mdns::CommssionAdvertiseMode mode)
-CHIP_ERROR Advertise(bool commissionableNode)
-{
     auto advertiseParameters = chip::Mdns::CommissionAdvertisingParameters().SetPort(CHIP_PORT).EnableIpV4(true);
     advertiseParameters.SetCommissionAdvertiseMode(commissionableNode ? chip::Mdns::CommssionAdvertiseMode::kCommissionableNode
                                                                       : chip::Mdns::CommssionAdvertiseMode::kCommissioner);
 
-    // TODO: device can re-enter commissioning mode after being fully provisioned
-    // (additionalPairing == true)
-    bool notYetCommissioned = !DeviceLayer::ConfigurationMgr().IsFullyProvisioned();
-    bool additionalPairing  = false;
-    advertiseParameters.SetCommissioningMode(notYetCommissioned, additionalPairing);
+    advertiseParameters.SetCommissioningMode(commissioningMode, additionalCommissioning);
```

Comment:
We fix the 'setCommisioningMode' to also not take several bools as arguments. Like split to 'EnableCommisioning' and 'EnableAdditionalCommisioning'. I am unclear why these assignment operations are made into a single call. `SetFoo(true, false)` is much less readable than `SetBar(true); SetBaz(false);`

I realize this is existing code, however given small change I would rather boy/girl-scout approach here and fix the advertise parameter awkwardness as  well. Multi-bool parameters make my code review  harder (I have to validate that the order of bool params matches the order that the header declares and uses them).

Reply:
fixed